src/Compilers/Core/CodeAnalysisTest/PEWriter/BlobTests.cs

```diff
@@ -856,5 +856,91 @@ public void Pooled()
             builder1.Free();
             builder2.Free();
         }
+
+        /// <summary>
+        /// This stream requires at least two calls to Read() to read all elements.
+        /// </summary>
+        private class TwoReadStream : Stream
+        {
+            public static byte[] BackingData { get; } = { 1, 2, 3, 4, 5 };
+
+            public override bool CanRead
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override bool CanSeek
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override bool CanWrite
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override long Length => BackingData.Length;
+
+            public override long Position { get; set; } = 0;
+
+            public override void Flush()
+            {
+                throw new NotImplementedException();
+            }
+
+            private bool _firstRead = true;
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+
+                // No matter what the client asks for, only give them half the first time
+                // Unless they only ask for 1 byte
+                if (_firstRead)
+                {
+                    count = (int)Math.Ceiling(count / 2.0);
```

Comment:
It could be, but it isn't in the test. I see no reason to be resilient to future consumers, since I anticipate no future consumers.